### ISSUE TITLE
VAT Obligations disappearing in production once submitted.

### ISSUE THREAD

**Issue by markogrady** [hmrc/vat-api#586]
Sorry to use this as a forum, but I just want to find out whether anyone else is experiencing a problem.

I have had 2 clients now who have submitted VAT returns, I receive a receipt etc.. Then when I go to get obligations and the obligation has disappeared.

**Comment by TS-grantlebrun** [comment]
What response are you getting from the API on the Obligations call when this happens? 
e.g. NOT_FOUND or CLIENT_OR_AGENT_NOT_AUTHORISED 

We have at least one customer that is seeing CLIENT_OR_AGENT_NOT_AUTHORISED now when trying to get Obligations after having filed a return successfully previously. Even re-authorising has not helped.

**Comment by markogrady** [comment]
I can see other returns for the client just not the one that was there previously and if I try to get the specific return I receive a 404. The auth token looks fine. I have sent a message to SDS support but sometimes it's easier to discuss on here.

**Comment by markogrady** [comment]
My client who is an agent has spoken to HMRC and they said The return has gone to the old database but not the new one.  They said client is in a transitional state where future returns should go in to the new side, which doesn't really help me because I don't mark the VAT return as fulfilled unless it appears as fulfilled in the get obligations API. I can change the code but it removes an additional layer of validation.

**Comment by cb-rfk** [comment]
Hi @markogrady. Thanks for posting that feedback.  

That response from HMRC really winds me up as we've been told so many times that the old system and the new system are completely separate entities and once a client completes their registration for MTD, they will exclusively be on the new system.  If they end up in transitional states across both, this will really impact the way we onboard users via our application, similar to what you say.

**Comment by nickcable** [comment]
I'm not sure if this is the same issue but we are getting a problem where a Customer has submitted a VAT return and a received a submission ref (receipt id) but the obligation is still there but shows incorrectly as status= 'Open' BUT also has a received date = start date of the period which is not correct either.  Has anybody else seen any of these symptons please ? I'm waiting for SDS support to respond. Confused.com

**Comment by sonicmule** [comment]
We have the same problem with one customer.
GetVatObligations returned the correct open VAT period.
Customer submitted their VAT Return.
GetVatObligations still showed the VAT Return as open.
A couple of days later, GetVatObligations no longer shows the submitted VAT Return but manages to list a whole load of VAT Returns from the old system.
Also, one would expect the new VAT period to be shown as open when your end date is 03-May-2019?

**Comment by junglistloydee** [comment]
We are also experiencing this. After an obligation has been uploaded, when you get obligations again you get a NOT_FOUND error.

**Comment by OriginalQuantum** [comment]
Yes this has come up a few times now for our customer's needs fixing asap thanks

**Comment by sparkz42** [comment]
I just had this issue as well. Customer submitted a return which was acknowledged ok (got bundle number). 
Checking obligations afterwards, the status hadn't changed but we were getting data ok. 
The next day though, checking obligations again we get 404 - NOT_FOUND.

**Comment by dronya** [comment]
Hi All,
I have a client with same problem. They send vat submit and recieve bundle_number, status ok and etc. But this doesn't appear in HMRC site: The obligation is still open.
Guys, how did you resolve this SDSTeam helped you or MTD VAT helpline support?

**Comment by sparkz42** [comment]
@dronya, does your client have a numeric PeriodKey and are you getting 404 - NOT_FOUND when checking on the obligation status?
 
The issue I had affected some clients with numeric PeriodKeys. It seems these users have their data (all or some) stored on a different location at HMRC for some reason. After successfully submitting a return, subsequent checks on the obligation just got a 404. It was an issue reported in the SDSTeam emails a while back.

HMRC did say they had fixed this in July and for our users it was certainly ok after that. It may be worth checking with the SDSTeam to see if the problem is back.

**Comment by dronya** [comment]
@sparkz42 my users do have numeric periodkey. After submission they can query the obligation using get vat apis for entering start date and end date not 404. But I will ask to do it again on Monday.
I also noticed that they were signed off for MTD after the reporting period was ended.
